### DISCUSSION TITLE
Optimizers disappeared

### DISCUSSION THREAD

**Discussion by Carlett86** [wlcrs/huawei_solar#775] [Setting up the integration]
For 2 months I have no longer been able to see the optimizer sensors, all of which are unavailable (until then I could see them without problems). I don't remember if it coincides with an update. now, having redownloaded the latest version of Huawei Solar, I no longer find the optimizers as "devices"

Describe your Huawei Solar Setup
Inverter Type: SUN2000-6KTL-L1
Inverter Firmware version: V200R001C00SPC141
sDongle present: Yes
sDongle Type: sDongleA-05 ( Ethernet)
sDongle Connectivitiy: Ethernet
sDongle Firmware: V200R022C10SPC114
Power meter present: single phase
Optimizers Present: Yes
Battery: LUNA2000-SO 20kWh
Battery Firmware version: V100R002C00SPC621
Huawei Solar integration version: 1.4.1

How do you connect to the inverter?
Via the SDongle, wired connection

Questo errore ha avuto origine da un'integrazione personalizzata.

Registratore: custom_components.huawei_solar
Fonte: custom_components/huawei_solar/__init__.py:190
Integrazione: Huawei Solar (documentazione, problemi)
Prima occorrenza: 11:41:49 (1 occorrenze)
Ultima registrazione: 11:41:49

Cannot create optimizer sensor entities due to an unexpected error
Traceback (most recent call last):
  File "/config/custom_components/huawei_solar/update_coordinator.py", line 94, in _async_update_data
    return await self.bridge.get_latest_optimizer_history_data()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/huawei_solar/bridge.py", line 379, in get_latest_optimizer_history_data
    file_data = await self._read_file(
                ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/huawei_solar/bridge.py", line 353, in _read_file
    return await self.client.get_file(file_type, customized_data, self.slave_id)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py", line 561, in get_file
    return await _do_read_file()
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py", line 548, in _do_read_file
    raise ReadException(
huawei_solar.exceptions.ReadException: Computed CRC ddda for file 68 does not match expected value 53462

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 312, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/huawei_solar/update_coordinator.py", line 96, in _async_update_data
    raise UpdateFailed(
homeassistant.helpers.update_coordinator.UpdateFailed: Could not update HV2270509907 optimizer values: Computed CRC ddda for file 68 does not match expected value 53462

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/huawei_solar/__init__.py", line 190, in async_setup_entry
    await create_optimizer_update_coordinator(
  File "/config/custom_components/huawei_solar/update_coordinator.py", line 118, in create_optimizer_update_coordinator
    await coordinator.async_config_entry_first_refresh()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 284, in async_config_entry_first_refresh
    raise ex
homeassistant.exceptions.ConfigEntryNotReady: Could not update HV2270509907 optimizer values: Computed CRC ddda for file 68 does not match expected value 53462

},
"huawei_solar": {
"documentation": "https://github.com/wlcrs/huawei_solar/wiki",
"version": "1.4.1",
"requirements": [
"huawei-solar==2.3.0"
]
},
"scheduler": {
"documentation": "https://github.com/nielsfaber/scheduler-component",
"version": "v0.0.0",
"requirements": []
}
},
"integration_manifest": {
"domain": "huawei_solar",
"name": "Huawei Solar",
"after_dependencies": [
"usb"
],
"config_flow": true,
"documentation": "https://github.com/wlcrs/huawei_solar/wiki",
"issue_tracker": "https://github.com/wlcrs/huawei_solar/issues",
"requirements": [
"huawei-solar==2.3.0"
],
"codeowners": [
"wlcrs"
],
"iot_class": "local_polling",
"version": "1.4.1",
"loggers": [
"huawei_solar",
"pymodbus"
],
"is_built_in": false
},
"setup_times": {},
"data": {
"config_entry_data": {
"host": "192.168.1.250",
"port": 502,
"slave_ids": [
1
],
"enable_parameter_configuration": true,
"username": null,
"password": null
},
"pymodbus_version": "3.6.8",
"slave_0": {
"model_name": "SUN2000-6KTL-L1",
"firmware_version": "V200R001C00",
"software_version": "V200R001C00SPC141",
"pv_string_count": 2,
"has_optimizers": 20,
"battery_type": 2,
"battery_1_type": 2,
"battery_2_type": 2,
"power_meter_type": 0,
"supports_capacity_control": true
},
"slave_0_inverter_data": {
"state_1": [
[
"Grid-Connected",
"Grid-Connected normally"
],
null
],
"state_2": [
[
"Unlocked",
"PV connected",
"DSP data collection"
],
null
],
"state_3": [
[
"On-grid",
"Off-grid switch disabled"
],
null
],
"alarm_1": [
[],
null
],
"alarm_2": [
[],
null
],
"alarm_3": [
[],
null
],
"pv_01_voltage": [
350.4,
"V"
],
"pv_01_current": [
7.07,
"A"
],
"pv_02_voltage": [
421.6,
"V"
],
"pv_02_current": [
9.22,
"A"
],
"input_power": [
6296,
"W"
],
"day_active_power_peak": [
6000,
"W"
],
"active_power": [
2550,
"W"
],
"efficiency": [
100.0,
"%"
],
"device_status": [
"On-grid",
null
],
"startup_time": [
"2024-06-30T04:18:10+00:00",
null
],
"shutdown_time": [
null,
null
],
"accumulated_yield_energy": [
16417.54,
"kWh"
],
"total_dc_input_power": [
17917.61,
"kWh"
],
"hourly_yield_energy": [
2.07,
"kWh"
],
"daily_yield_energy": [
24.98,
"kWh"
],
"nb_online_optimizers": [
20,
null
]
},
"slave_0_power_meter_data": {
"power_meter_active_power": [
2116,
"W"
],
"active_grid_power_factor": [
0.968,
null
],
"grid_exported_energy": [
7903.47,
"kWh"
],
"grid_accumulated_energy": [
3597.84,
"kWh"
]
},
"slave_0_battery_data": {
"storage_state_of_capacity": [
22.0,
"%"
],
"storage_running_status": [
2,
null
],
"storage_bus_voltage": [
478.0,
"V"
],
"storage_bus_current": [
7.7,
"A"
],
"storage_charge_discharge_power": [
3745,
"W"
],
"storage_total_charge": [
607078.34,
"kWh"
],
"storage_total_discharge": [
6548.11,
"kWh"
],
"storage_current_day_charge_capacity": [
3.23,
"kWh"
],
"storage_current_day_discharge_capacity": [
13.07,
"kWh"
]
},
"slave_0_config_data": {
"active_power_fixed_value_derating": [
60,
"W"
],
"storage_maximum_charging_power": [
3720,
"W"
],
"storage_maximum_discharging_power": [
5000,
"W"
],
"storage_charging_cutoff_capacity": [
100.0,
"%"
],
"storage_discharging_cutoff_capacity": [
5.0,
"%"
],
"storage_forced_charging_and_discharging_period": [
0,
"min"
],
"storage_working_mode_settings": [
2,
null
],
"storage_charge_from_grid_function": [
true,
null
],
"storage_grid_charge_cutoff_state_of_charge": [
20.0,
"%"
],
"forcible_charge_discharge_write": [
0,
null
],
"storage_forcible_charge_discharge_soc": [
0.0,
"%"
],
"storage_fixed_charging_and_discharging_periods": [
[],
null
],
"storage_power_of_charge_from_grid": [
0,
"W"
],
"storage_maximum_power_of_charge_from_grid": [
3000,
"W"
],
"storage_forcible_charge_discharge_setting_mode": [
0,
null
],
"storage_forcible_charge_power": [
0,
null
],
"storage_forcible_discharge_power": [
0,
null
],
"storage_time_of_use_charging_and_discharging_periods": [
[
{
"__type": "<class 'huawei_solar.registers.HUAWEI_LUNA2000_TimeOfUsePeriod'>",
"repr": "HUAWEI_LUNA2000_TimeOfUsePeriod(start_time=0, end_time=1439, charge_flag=<ChargeFlag.DISCHARGE: 1>, days_effective=(True, True, True, True, True, True, True))"
}
],
null
],
"storage_excess_pv_energy_use_in_tou": [
0,
null
],
"storage_capacity_control_mode": [
1,
null
],
"storage_capacity_control_soc_peak_shaving": [
5.0,
"%"
],
"storage_capacity_control_periods": [
[
{
"__type": "<class 'huawei_solar.registers.PeakSettingPeriod'>",
"repr": "PeakSettingPeriod(start_time=0, end_time=1439, power=5000, days_effective=(True, True, True, True, True, True, True))"
}
],
null
]
}
}
}

Registratore: huawei_solar.huawei_solar
Fonte: /usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py:203
Prima occorrenza: 11:40:36 (1 occorrenze)
Ultima registrazione: 11:40:36

Aborting client creation due to error.
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py", line 199, in create
    await huawei_solar._initialize()
  File "/usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py", line 118, in _initialize
    self.time_zone = (await self.get(rn.TIME_ZONE)).value
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py", line 271, in get
    return (await self.get_multiple([name], slave))[0]
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py", line 308, in get_multiple
    response = await self._read_registers(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py", line 451, in _read_registers
    return await _do_read()
           ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/backoff/_async.py", line 151, in retry
    ret = await target(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/backoff/_async.py", line 151, in retry
    ret = await target(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py", line 399, in _do_read
    response = await self._client.read_holding_registers(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pymodbus/client/base.py", line 183, in async_execute
    raise ModbusIOException(
pymodbus.exceptions.ModbusIOException: Modbus Error: [Input/Output] ERROR: No response received after 3 retries

Please confirm the following:
 I'm running the latest release of Home Assistant.
 I'm running the latest release of this integration.
 I did not find an existing issue describing this problem.
 I did upload the diagnostics-file that I could retrieve from the 'Devices & Services Page'
 I did upload the relevant debug logs (via 'Enable Debug Logging'-feature or by manually configuring HA logging)
![Ottimizzatore](https://github.com/user-attachments/assets/6676d0b8-026e-48ef-8d42-cbee0bb005de)

**Comment by cangaroo82** [comment]
I can't see them
did you find the solution
